### PR TITLE
Fixed #29

### LINKED ISSUE #29: Add Grid component

### PR DESCRIPTION
**Pull request by binoy14**
Adds Grid Component with same usage API as https://github.com/GeekyAnts/react-native-easy-grid

### CODE REVIEW

File: src/grid/Grid.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import React, { Component, PropTypes } from 'react';
+import { View } from 'react-native';
+import Row from "./Row";
```
Comment (Monte9) [inline src/grid/Grid.js]: use single quotes (' ') to keep it consistent

File: src/grid/Grid.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  static propTypes = {
+    style: PropTypes.object,
+  }
+
+  isRow() {
+    let isRow = false;
+    React.Children.forEach(this.props.children, (child) => {
+      if (child.type === Row) {
+        console.log('here')
```
Comment (Monte9) [inline src/grid/Grid.js]: remove console.log

File: src/grid/Col.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React, { PropTypes } from "react";
+import { View } from "react-native";
+
+const Col = (props) => {
+	const {style, size, width} = props;
```
Comment (Monte9) [inline src/grid/Col.js]: indentation is off?

File: src/grid/Row.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import React, { PropTypes } from "react";
+import { View } from "react-native";
+
+const Row = (props) => {
+	const {style, size} = props;
```
Comment (Monte9) [inline src/grid/Row.js]: indentation is off?

File: src/index.js [hunk 5]
```diff
@@ -18,6 +18,9 @@ import Tab from './tabs/Tab'
 import colors from './config/colors'
 import getIconType from './helpers/getIconType'
 import normalize from './helpers/normalizeText'
+import Grid from "./grid/Grid"
```
Comment (Monte9) [inline src/index.js]: use single quotes (' ') to stay consistent

**Comment by Monte9** [comment]
left a few minor styling comments. Will try the grid layout soon and implement it in the hackathon-starter project as a demo. 
Also @binoy14, could you kindly change the base branch as [v0.9.0](https://github.com/react-native-community/react-native-elements/tree/v0.9.0). Thanks.

File: Readme.MD [hunk 6]
```diff
@@ -945,3 +945,79 @@ import { PricingCard } from 'react-native-elements'
...
+
+![Nested Layout](http://i.imgur.com/PZ4vR6z.png)
+
+```js
+import {Grid, Col, Row} from 'react-native-elements';
+
+<Grid>
+  <Col></Col>
+  <Col>
+  <Row></Row>
```
Comment (Monte9) [inline Readme.MD]: indentation is off.

**Comment by Monte9** [comment]
@binoy14 there is an outstanding PR on the react-native-easy-grid repo that I think would be a great addition. Check this out: https://github.com/GeekyAnts/react-native-easy-grid/pull/21

What are your thoughts on adding the onPress feature to this PR?

File: Readme.MD [hunk 7]
```diff
@@ -945,3 +945,79 @@ import { PricingCard } from 'react-native-elements'
...
+
+<Grid>
+  <Col></Col>
+  <Col>
+    <Row></Row>
+    <Row></Row>
+  </Col>
+</Grid>
+```
+
```
Comment (Monte9) [inline Readme.MD]: Consider adding an example showing the use of `size` props.

File: Readme.MD [hunk 8]
```diff
@@ -945,3 +945,79 @@ import { PricingCard } from 'react-native-elements'
...
+<Grid>
+  <Row></Row>
+  <Row></Row>
+</Grid>
+```
+
+#### Column
+
+![Column Component](http://i.imgur.com/MWIBLKk.png)
```
Comment (Monte9) [inline Readme.MD]: The image looks really big. Consider using the `` tag as that let's you control the size of the image. 
``

**Comment by binoy14** [comment]
@Monte9 What's the advantage/purpose of adding an onPress? I don't mind adding it, just trying to understand how will it be useful.

**Comment by Monte9** [comment]
@binoy14 Although the onPress does not directly improve the functionality of the grid layout itself, I think having it will give the user some flexibility in case they need to use it. One use case I can think of is.. Using flex to position an image but then having to wrap it in a TouchableHighlight to make it clickable. In this case, if I am using the grid layout, I don't have to wrap the image and I can directly use the onPress of the Col/Row. Does that make sense?

Also I just tried the Grid layout from this PR and I really like it. Good job! 

screen shot 2016-12-27 at 3 58 22 pm

**Comment by binoy14** [comment]
@Monte9 I added onPress event for all three elements.

I have set default activeOpacity to 1. It made sense to have default of one and give option to change it by the consumer. If you would like me to change that let me know.

**Comment by dabit3** [comment]
@binoy14 @Monte9 is this PR ready to merge?

**Comment by binoy14** [comment]
@dabit3 from my side yes

**Pull request merged**